### PR TITLE
An implementation for openstack Swift

### PR DESCRIPTION
**Pull request by mawinter69**
An implementation of the artifact manager plugin with jclouds for openstack swift.

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/artifact_manager_jclouds/swift/SwiftBlobstore.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+	public BlobStore getBlobStore() throws IOException {
+		return getRegionContext().getBlobStore(getRegion());
+	}
+
+	@Override
+	public URI toURI(String container, String key) {
+		try {
+			Blob blob = getBlobStore().blobBuilder(key).build();
+			blob.getMetadata().setContainer(container);
+			return toExternalURL(blob, HttpMethod.GET).toURI();
```
Comment (jglick) [inline src/main/java/io/jenkins/plugins/artifact_manager_jclouds/swift/SwiftBlobstore.java]: This should not be implemented using a presigned URL. It should be a predictable URI without authentication information.

Comment (mawinter69) [reply src/main/java/io/jenkins/plugins/artifact_manager_jclouds/swift/SwiftBlobstore.java]: Not sure if this is possible.
You can access the blobstore directly only when you have the corresponding permissions configured I think, so no anonymous access.

Comment (jglick) [reply src/main/java/io/jenkins/plugins/artifact_manager_jclouds/swift/SwiftBlobstore.java]: The URI need not be a resolvable URL. It is not in the case of the S3 (incl. MinIO) implementation. [Javadoc](https://javadoc.jenkins.io/jenkins/util/VirtualFile.html#toURI--)

Comment (jglick) [reply src/main/java/io/jenkins/plugins/artifact_manager_jclouds/swift/SwiftBlobstore.java]: (probably the method in this plugin’s API should `@link` to that for clarity)

**Pull request closed without merging**